src/SD_calculator.jl

```diff
@@ -0,0 +1,35 @@
+"""
+    getSD(A::AbstractArray)
+
+If your image is not uniformily sampled, use this to get the `SD` matrix, which represents spacing along all axes of an image.
+Please remember to strip out the time axis.
+
+# Examples
+```julia-repl
+julia> myimage
+Normed ImageMeta with:
+  data: 3-dimensional AxisArray{N2f14,3,...} with axes:
+    :x, 0.0 μm:0.71 μm:6.39 μm
+    :l, 0.0 μm:0.71 μm:6.39 μm
+    :z, 0.0 μm:6.2 μm:55.8 μm
+And data, a 10×10×10 Array{N2f14,3} with eltype Normed{UInt16,14}
+  properties:
+    imagineheader: <suppressed>
+
+julia> getSD(myimage)
+3×3 SArray{Tuple{3,3},Float64,2,9} with indices SOneTo(3)×SOneTo(3):
+ 0.71  0.0   0.0
+ 0.0   0.71  0.0
+ 0.0   0.0   6.2
+```
+"""
+function getSD(A::AbstractArray{T,N}) where {T,N} #only works on non-skewed arrays at the momoent.
+    sd = spacedirections(A)
+    SD = zeros(N,N)
+    for i = 1:N
+        for j = 1:N
+            SD[i,j] = sd[i][j]/oneunit(sd[i][j])

```

Comment:
Here's a problem:
```julia
using Unitful: μm, mm, s

img = AxisArray(rand(3, 4, 5), (:x, :y, :time), (5μm, 1mm, 0.3s))
sd = spacedirections(img)
SD = zeros(2, 2)
for i = 1:2, j = 1:2
    SD[i,j] = sd[i][j]/oneunit(sd[i][j])
end

julia> SD
2×2 Array{Float64,2}:
 5.0  0.0
 0.0  1.0
```
So `5μm` is 5x bigger than `1mm`?

Contrast:
```julia
minps = minimum(pixelspacing(img))
SD = zeros(2, 2)
for i = 1:2, j = 1:2
    SD[i,j] = sd[i][j]/minps
end

julia> SD
2×2 Array{Float64,2}:
 1.0    0.0
 0.0  200.0
```
which is the answer you want here.

Generalizable principle: when dealing with units, don't think like a computer scientist (i.e., "what type is this?"), think like a physicist ("what operations are meaningful and have intrinsic physical reality?")

Reply:
I don't think I'm supposed to be running into this issue:

```
julia> (1u"mm")/(1u"μm")
1.0 mm μm^-1
```